### PR TITLE
feat: UI Implemenatation of raffle code generator

### PR DESCRIPTION
**Pull request by cheedinmar**

### CODE REVIEW

**Review by darmhoo — COMMENTED** [review]
It looks good

**Review by darmhoo — APPROVED** [review]

File: src/components/CopyItem.vue [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    Flash.setSuccess(`${this.item} has been copied to clipboard!`);
+                })
+                .catch((error) => {
+                    Flash.setError("Copy failed. Please try again.")
+                });
+        },
+    },
+};
+</script>
+<style>
```
Comment (teebarg) [inline src/components/CopyItem.vue:39]: let's add a scope to the style tag to avoid leakage

File: src/components/PaymentLog.vue [hunk 2]
```diff
@@ -710,6 +734,22 @@ export default {
...
+    },
+    async validateRaffleCode(){
+      await post(this.apiUrls.raffle_code, {
+           "phone_number": this.customer.telephone,
+            "code": this.salesLogForm.raffle_code
+        }).then(res=>{
+           Flash.setSuccess('Raffle Code Validated')
+          this.validatedRaffleCode = true
+        }).catch(error =>{
+           this.error = error.response.data.message
```
Comment (teebarg) [inline src/components/PaymentLog.vue:749]: Can we use optional chaining?

File: src/views/Admin/ReportingTool.vue [hunk 3]
```diff
@@ -482,12 +496,14 @@
 				};
 			},
 			async getReport() {
+				console.log(this.billboardOrders, 'from get reports')
```
Comment (teebarg) [inline src/views/Admin/ReportingTool.vue:499]: remove console

File: src/views/Admin/raffle_code_generator/raffle_code_generator.vue [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+			raffle_code:'ATfy65GO'
+		}
+	},
+	methods: {
+		async GenerateRaffleCode(){
+			this.formatPhoneNumber()
+			 await post('/api/raffle_draw_code', {
+				phone_number: this.phone_number
+			}).then(res=>{
+				console.log(res)
```
Comment (teebarg) [inline src/views/Admin/raffle_code_generator/raffle_code_generator.vue:56]: remove console

File: src/views/Admin/raffle_code_generator/raffle_code_generator.vue [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+				this.raffle_code = res.data.data.code
+				this.result = true
+			}).catch(err=>{
+				 Flash.setError('This number has a code that has not been used')
+				console.log(err)
```
Comment (teebarg) [inline src/views/Admin/raffle_code_generator/raffle_code_generator.vue:61]: remove console

**Review by teebarg — APPROVED** [review]

**Pull request merged**